docs/api/openapi.yaml

```diff
@@ -0,0 +1,217 @@
+        '400':
+          description: Invalid URN:NBN identifier supplied
+        '401':
+            $ref: '#/components/responses/UnauthorizedError'      
+        '404':
+          description: Supplied URN:NBN identifier not found, or nbn-prefix not registered to this user.
+      tags:
+        - URN:NBN identifier
+
+    put:
+      summary: 'Updates an existing URN:NBN {identifier} or registers a new URN:NBN {identifier} associated with a prioritized list of locations.'
+      description: 'Updates or registers a (new) URN:NBN associated with a prioritized list of locations. Multiple locations are prioritized in respective order. The first location is the preferred. The second is failover, etc. <br>Identifiers must have a prefix that matches the authenticated user.'
+      operationId: 'updateNbnRecord'
+      requestBody:
+        required: true
+        description: A json object that contains the assosiated locations for the {identifier}.
+        content:
+          application/json:
+            schema:
+              type: array
+              items:
+                $ref: '#/components/schemas/Location'
+      responses:
+        '200':
+          description: OK (updated existing)
+        '201':
+          description: Successful operation (created new)          
+        '400':
+          description: Invalid URN:NBN identifier or location(s) supplied
+        '401':
+            $ref: '#/components/responses/UnauthorizedError'
+      tags:
+        - URN:NBN identifier
+        
+  /nbn/{identifier}/location:
+    parameters:
+      - $ref: '#/components/parameters/nbn_identifier'
+    get:
+      summary: 'Returns registered locations for this URN:NBN {identifier}.'
+      description: 'Returns all registed locations for this {identifier}.<br>The identifier must have a prefix that matches the authenticated user. HTTP 404 otherwise.'
+      operationId: 'getLocationsByNbn'
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array              
+                items:
+                  $ref: '#/components/schemas/Location'
+        '400':
+          description: Invalid URN:NBN identifier supplied
+        '401':
+            $ref: '#/components/responses/UnauthorizedError'  
+        '404':
+          description: Supplied URN:NBN identifier not found
+      tags:
+        - URN:NBN identifier
+
+  
+  /location/{location}:
+    parameters:
+    - $ref: '#/components/parameters/uri_location'
+    get:
+      operationId: getNbnByLocation
+      summary: Returns URN:NBN identifier registered for this location.
+      description: Returns URN:NBN identifier associated with this location.'   
+      responses:
+        '200':
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NbnIdentifier'
+          description: OK
+        '401':
```

Comment:
it only returns NbnIdentifiers from this authenticated user?

Reply:
Good question: I think that it should return all identifiers, if at least one urn:nbn is registered with the authenticated user (IR).